pkg/operator/helpers/helpers.go

```diff
@@ -451,14 +452,15 @@ func DetermineReplica(ctx context.Context, kubeClient kubernetes.Interface, mode
 		}
 	}
 
-	return DetermineReplicaByNodes(ctx, kubeClient)
+	return DetermineReplicaByNodes(ctx, kubeClient, masterNodeLabelSelector)
 }
 
 // DetermineReplicaByNodes determines the replica of deployment based on:
 // list master nodes in the cluster and return 1 if
 // the number of master nodes is equal or less than 1. Return 3 otherwise.
-func DetermineReplicaByNodes(ctx context.Context, kubeClient kubernetes.Interface) int32 {
-	nodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master="})
+func DetermineReplicaByNodes(ctx context.Context, kubeClient kubernetes.Interface, masterNodeLabelSelector map[string]string) int32 {
+	labelSelector := mapToString(masterNodeLabelSelector)
```

Comment:
suggest to use `FormatLabelSelector`: https://github.com/open-cluster-management-io/ocm/blob/92a2c91f15947d30627bf2652c5155cf81971437/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go#L169

Reply:
updated